### PR TITLE
Feature/tables

### PR DESCRIPTION
**Pull request by francocava**

### CODE REVIEW

File: src/api/assignments/index.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import axios from "../../plugins/axios";
+
+class Assignments {
```
Comment (agustinfranc) [inline src/api/assignments/index.ts]: export class AssignmentsClient / AssignmentsApi

File: src/api/assignments/index.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    try {
+      const response = await axios.get("/assignments");
+      return response;
+    } catch (error) {
+      return error;
+    }
+  }
+}
+
+export default Assignments;
```
Comment (agustinfranc) [inline src/api/assignments/index.ts]: remove line

File: src/components/tables/genericTable.vue [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      filter: true,
+      flex: 1,
+    };
+
+    // Example load data from sever
+    /* onMounted(() => {
+      fetch("https://www.ag-grid.com/example-assets/row-data.json")
+        .then((result) => result.json())
+        .then((remoteRowData) => (rowData.value = remoteRowData));
+    }); */
```
Comment (agustinfranc) [inline src/components/tables/genericTable.vue]: remover codigo comentado

File: src/components/tables/genericTable.vue [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+      },
+      deselectRows: () => {
+        gridApi.value.deselectAll();
+      },
+    };
+  },
+});
+</script>
+
+<style lang="scss"></style>
```
Comment (agustinfranc) [inline src/components/tables/genericTable.vue]: remover linea

File: src/views/AssignmentsView.vue [hunk 5]
```diff
@@ -1,18 +1,39 @@
 <template>
-  <Assignments />
+  <LazyTable :columns="columns" @request="getAssignments" />
 </template>
 
 <script lang="ts">
 import { defineComponent } from "vue";
+import Assignemts from "../api/assignments/index";
```
Comment (agustinfranc) [inline src/views/AssignmentsView.vue]: import AssignmentClient / AssignmentApi

File: src/api/assignments/index.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import axios from "../../plugins/axios";
+
+class Assignments {
+  static async get() {
+    try {
+      const response = await axios.get("/assignments");
```
Comment (agustinfranc) [inline src/api/assignments/index.ts]: return await ...

File: src/api/assignments/index.ts [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+      return response;
```
Comment (agustinfranc) [inline src/api/assignments/index.ts]: remove line

File: src/api/assignments/index.ts (hunk 2 above)
Comment (francocava) [reply src/api/assignments/index.ts]: la 13?

File: src/api/assignments/index.ts (hunk 7 above)
Comment (francocava) [reply src/api/assignments/index.ts]: which one?

File: src/api/assignments/index.ts (hunk 6 above)
Comment (francocava) [reply src/api/assignments/index.ts]: where is the return?

File: src/api/assignments/index.ts (hunk 2 above)
Comment (agustinfranc) [reply src/api/assignments/index.ts]: 14

File: src/api/assignments/index.ts (hunk 6 above)
Comment (agustinfranc) [reply src/api/assignments/index.ts]: que reemplaces la linea 6 por: return await axios.get("/assignments");

File: src/api/assignments/index.ts (hunk 7 above)
Comment (agustinfranc) [reply src/api/assignments/index.ts]: 7, siempre es la ultima que se ve

**Pull request merged**